itinerant-iterators/app/steganography/steganography.py

```diff
@@ -0,0 +1,123 @@
+from functools import cached_property
+from pathlib import Path
+from typing import Protocol
+
+import PIL
+from PIL.Image import Image as PILImage
+
+
+class Image:
+    """Image represents images to be encoded with data using steganography"""
+
+    def __init__(self, filename: Path):
+        self.__filename = filename
+        self.__image = PIL.Image.open(str(filename)).convert('RGB')
+
+    def __iter__(self):
+        """Iterate over the image's pixels."""
+        return iter(self.image.getdata())
+
+    @property
+    def image(self) -> PILImage:
+        """Get the image"""
+        return self.__image
+
+    @cached_property
+    def max_bytes(self) -> int:
+        """Maximum number of bytes that can be encoded in the image"""
+        shape = self.__image.size
+        return (shape[0] * shape[1] * 3) // 8
+
+    def can_encode(self, b: bytes) -> bool:
+        """Check whether the given bytes can be encoded in the image
+
+        Args:
+            b (bytes): bytes to be encoded in the image
+
+        Returns:
+            bool: whether the given bytes can be encoded in the image
+        """
+        return len(b) <= self.max_bytes
+
+    @property
+    def pixels(self) -> list[list[int]]:
+        """Get the pixels of the image
+
+        Returns:
+            list[list[int]]: Array of pixels in the image, where each pixel is a list of integers
+        """
+        return list([list(rgb) for rgb in self.image.getdata()])
```

Comment:
List comprehensions, as their name suggests, build a new list. No need to call list again.

```suggestion
        return [list(rgb) for rgb in self.image.getdata()]]
```